src/Compilers/CSharp/Portable/Binder/Binder_AnonymousTypes.cs

```diff
@@ -84,8 +84,11 @@ private BoundExpression BindAnonymousObjectCreation(AnonymousObjectCreationExpre
 
                 // build anonymous type field descriptor
                 fieldSyntaxNodes[i] = (nameToken.Kind() == SyntaxKind.IdentifierToken) ? (CSharpSyntaxNode)nameToken.Parent : fieldInitializer;
-                fields[i] = new AnonymousTypeField(fieldName == null ? "$" + i.ToString() : fieldName, fieldSyntaxNodes[i].Location, 
-                                                   TypeSymbolWithAnnotations.Create(fieldType, isNullableIfReferenceType: node.IsFeatureStaticNullCheckingEnabled()));
+                bool? isNullableIfReferenceType = ((CSharpParseOptions)node.SyntaxTree?.Options)?.IsFeatureEnabled(MessageID.IDS_FeatureStaticNullChecking) == true;
```

Comment:
>isNullableIfReferenceType [](start = 22, length = 25)

Not related to this PR: I don't think we should bind differently based on feature flag. We should probably always bind anonymous type members as null-oblivious and then consider revising during nullability analysis.
Could you update this code with a comment and a link to [https://github.com/dotnet/roslyn/issues/24018](https://github.com/dotnet/roslyn/issues/24018) ?